subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/AttributeMatchingConfigurationSelector.java

```diff
@@ -237,11 +253,11 @@ private static MatchResult containsImplicitCapability(CapabilitiesMetadata capab
         return MatchResult.NO_MATCH;
     }
 
-    private static boolean fastContainsImplicitCapability(CapabilitiesMetadataInternal capabilitiesMetadata, Collection<? extends Capability> capabilities) {
+    private boolean fastContainsImplicitCapability(CapabilitiesMetadataInternal capabilitiesMetadata, Collection<? extends Capability> capabilities) {
         return capabilities.isEmpty() || capabilitiesMetadata.isShadowedCapabilityOnly();
     }
 
-    private static Capability unwrap(Capability capability) {
```

Comment:
We've discussed this already. 😄👎  The general issues of 1) when are warnings acceptable and 2) is this warning useful or should it be removed/downgraded to a weak warning should be discussed elsewhere.

In this particular case, however, I'm not very concerned here as I don't see the potential for these methods to be static remaining for long.  The whole point of this type is to allow injection of "something" that can modify the failure messages.  That "something" will likely be a field these methods will reference or a new location where these methods will be relocated to.  It isn't worth worrying about this yet.

Reply:
To follow up here, I agree this is probably a waste of time for us to argue about here, but the fact remains that the IDE enforces the style that methods which do not depend on project instance variables should be marked `static`. 

If you intend to continue making changes like this, I strongly believe you should query the team and petition to remove the IDE check. Otherwise, we just get into a cyclical state where developers will see the IDE warnings and just reverse these changes.